third_party/gpus/cuda/BUILD.tpl

```diff
@@ -0,0 +1,235 @@
+    values = {
+        "define": "using_cuda_clang=true",
+        "compilation_mode": "opt",
+    },
+)
+
+config_setting(
+    name = "darwin",
+    values = {"cpu": "darwin"},
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "cuda_headers",
+    hdrs = glob([
+        "**/*.h",
+    ]),
+    includes = [".", "include"],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "cudart_static",
+    srcs = [
+        cuda_static_library_path("cudart"),
+    ],
+    includes = ["include/"],
+    linkopts = [
+        "-ldl",
+        "-lpthread",
+    ] + select({
+        "@//tensorflow:darwin": [],
+        "//conditions:default": ["-lrt"]
+    }),
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "cudart",
+    srcs = [
+        cuda_library_path("cudart")
+    ],
+    data = [
+        cuda_library_path("cudart")
+    ],
+    includes = ["include/"],
+    visibility = ["//visibility:public"],
+    linkstatic = 1,
+)
+
+cc_library(
+    name = "cublas",
+    srcs = [
+        cuda_library_path("cublas")
+    ],
+    data = [
+        cuda_library_path("cublas")
+    ],
+    includes = ["include/"],
+    visibility = ["//visibility:public"],
+    linkstatic = 1,
+)
+
+cc_library(
+    name = "cudnn",
+    srcs = [
+        cudnn_library_path()
+    ],
+    data = [
+        cudnn_library_path()
+    ],
+    includes = ["include/"],
+    visibility = ["//visibility:public"],
+    linkstatic = 1,
+)
+
+cc_library(
+    name = "cufft",
+    srcs = [
+        cuda_library_path("cufft")
+    ],
+    data = [
+        cuda_library_path("cufft")
+    ],
+    includes = ["include/"],
+    visibility = ["//visibility:public"],
+    linkstatic = 1,
+)
+
+cc_library(
+    name = "curand",
+    srcs = [
+        cuda_library_path("curand")
+    ],
+    data = [
+        cuda_library_path("curand")
+    ],
+    includes = ["include/"],
+    visibility = ["//visibility:public"],
+    linkstatic = 1,
+)
+
+cc_library(
+    name = "cuda",
+    deps = [
+        ":cuda_headers",
+        ":cudart",
+        ":cublas",
+        ":cudnn",
+        ":cufft",
+        ":curand",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "cupti_headers",
+    hdrs = glob([
+        "**/*.h",
+    ]),
+    includes = [".", "extras/CUPTI/include/"],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "cupti_dsos",
+    data = [
+        cupti_library_path(),
+    ],
+    visibility = ["//visibility:public"],
+)
+
+
+# TODO(opensource): for now, we have to invoke the cuda_config.sh manually in the source tree.
+# This rule checks if Cuda libraries in the source tree has been properly configured.
+# The output list makes bazel runs this rule first if the Cuda files are missing.
+# This gives us an opportunity to check and print a meaningful error message.
+# But we will need to create the output file list to make bazel happy in a successful run.
+#genrule(
```

Comment:
I was always considering this as a way to warn the user about cuda
configuration being unavailable. If cuda is deactivating, TF should fails
with a nice message when trying to build for cuda, right?

On Thu, Jul 28, 2016 at 12:17 PM David Z. Chen <notifications@github.com>
wrote:

> In third_party/gpus/cuda/BUILD.tpl
> <https://github.com/tensorflow/tensorflow/pull/3269#discussion_r72597490>:
>
> > +
> > +cc_library(
> > +    name = "cupti_dsos",
> > +    data = [
> > +        cupti_library_path(),
> > +    ],
> > +    visibility = ["//visibility:public"],
> > +)
> > +
> > +
> > +# TODO(opensource): for now, we have to invoke the cuda_config.sh manually in the source tree.
> > +# This rule checks if Cuda libraries in the source tree has been properly configured.
> > +# The output list makes bazel runs this rule first if the Cuda files are missing.
> > +# This gives us an opportunity to check and print a meaningful error message.
> > +# But we will need to create the output file list to make bazel happy in a successful run.
> > +#genrule(
>
> I'm not completely following. This is a sanity check to determine whether
> the cuda files are set up correctly. In the context of this change, we
> would want a similar sanity check to determine whether @local_config_cuda
> is reasonable, which, come to think of it, might be more suitable
> implemented in cuda_configure.bzl rather than as a build target. WDYT?
>
> —
> You are receiving this because you were mentioned.
>
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/3269/files/96a4da93ddddb723216a601f3a876b466f78558c#r72597490>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADjHf_gPZztewrxTPWKs1p872-5RQoGQks5qaIHIgaJpZM4JJPRy>
> .
>